server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -94,6 +96,8 @@ public CoordinatorDynamicConfig(
                                    ? current.getKillDataSourceWhitelist()
                                    : parseKillDataSourceWhitelist(killDataSourceWhitelist);
 
+    this.maxSegmentsInQueue = maxSegmentsInQueue == null ? current.getMaxSegmentsInQueue() : maxSegmentsInQueue;
```

Comment:
From where the `current.getMaxSegmentsInQueue()` value will come?

For backwards compatibility, I think maxSegmentsInQueue=0 should mean unlimited.